src/idl_parser.cpp

```diff
@@ -2576,10 +2580,18 @@ CheckedError Parser::ParseProtoFields(StructDef *struct_def, bool isextend,
           auto val = attribute_;
           ECHECK(ParseProtoCurliesOrIdent());
           if (key == "default") {
-            // Temp: skip non-numeric defaults (enums).
             auto numeric = strpbrk(val.c_str(), "0123456789-+.");
-            if (IsScalar(type.base_type) && numeric == val.c_str())
+            if (IsScalar(type.base_type) && numeric == val.c_str()) {
               field->value.constant = val;
+            } else if (type.enum_def) {
+              const EnumVal* ev = type.enum_def->Lookup(val);
+              if(!ev) {
```

Comment:
Yes, this check is redundant if we're OK relying on `protoc` to do correctness checks for the proto file.

Would you prefer that we emit a malformed `.fbs` file if the proto is invalid, rather than reporting the error?

Reply:
You are right. In any case, a null-pointer check is needed here.

Reply:
Most proto related code in this file assumes the `.proto` has gone thru `protoc` without errors, and there's probably a lot of places where it would give an unhelpful error or generate nonsensical code if you fed it a `.proto` with errors in it. I did that to keep the code simple, as at the time I didn't want to write a fully features `.proto` parser. That said, more correct code always welcome, though entirely up to you.